File: Baekjoon/Class2/24.11651 좌표 정렬하기 2.cpp
#include <iostream>
#include <utility>
#include <vector>
#include <algorithm>
using namespace std;

bool asc_yx(pair<int, int> c1, pair<int, int> c2) {
    if (c1.second != c2.second)
        return c1.second < c2.second;
    else
        return c1.first < c2.first;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    cin >> n;
    vector<pair<int, int>> coords(n);
    for (int i = 0; i < n; i++)
        cin >> coords[i].first >> coords[i].second;

    sort(coords.begin(), coords.end(), asc_yx);
    for (pair<int, int> coord : coords)
        cout << coord.first << ' ' << coord.second << '\n';
    return 0;
}

File: Baekjoon/Class2/24.11651 좌표 정렬하기 2.py
import sys
input = sys.stdin.readline

n = int(input())
coords = []
for _ in range(n):
    x, y = map(int, input().split())
    coords.append([x, y])

coords.sort(key = lambda coord: (coord[1], coord[0]))
print("\n".join([f"{coord[0]} {coord[1]}" for coord in coords]))